src/uwtools/tests/config/test_atparse_to_jinja2.py

```diff
@@ -18,83 +18,114 @@
 
 
 @fixture
-def atparselines():
-    return ["@[greeting] to the @[subject]", "@[flowers] are @[color]"]
+def txt_atparse():
+    return """
+@[greeting] to the @[subject]
+@[flowers] are @[color]
+""".strip()
 
 
 @fixture
-def atparsefile(atparselines, tmp_path):
-    path = tmp_path / "atparse.txt"
-    with open(path, "w", encoding="utf-8") as f:
-        for line in atparselines:
-            print(line, file=f)
-    return path
+def txt_jinja2():
+    return """
+{{ greeting }} to the {{ subject }}
+{{ flowers }} are {{ color }}
+""".strip()
 
 
 @fixture
-def jinja2txt():
-    return "{{greeting}} to the {{subject}}\n{{flowers}} are {{color}}\n"
+def atparsefile(txt_atparse, tmp_path):
+    path = tmp_path / "atparse.txt"
+    with open(path, "w", encoding="utf-8") as f:
+        print(txt_atparse, file=f)
+    return path
 
 
 # Test functions
 
 
-def test_convert_input_file_to_output_file(atparsefile, capsys, jinja2txt, tmp_path):
+def test_convert_input_file_to_output_file(atparsefile, capsys, txt_jinja2, tmp_path):
     outfile = tmp_path / "outfile"
     atparse_to_jinja2.convert(input_file=atparsefile, output_file=outfile)
     with open(outfile, "r", encoding="utf-8") as f:
-        assert f.read() == jinja2txt
+        assert f.read().strip() == txt_jinja2
     streams = capsys.readouterr()
     assert not streams.err
     assert not streams.out
 
 
-def test_convert_input_file_to_logging(atparsefile, caplog, capsys, jinja2txt, tmp_path):
+def test_convert_input_file_to_logging(atparsefile, caplog, capsys, txt_jinja2, tmp_path):
     log.setLevel(logging.INFO)
     outfile = tmp_path / "outfile"
     atparse_to_jinja2.convert(input_file=atparsefile, dry_run=True)
     streams = capsys.readouterr()
-    assert "\n".join(record.message for record in caplog.records) == jinja2txt.strip()
+    assert "\n".join(record.message for record in caplog.records).strip() == txt_jinja2
     assert not streams.out
     assert not outfile.is_file()
 
 
-def test_convert_input_file_to_stdout(atparsefile, capsys, jinja2txt):
+def test_convert_input_file_to_stdout(atparsefile, capsys, txt_jinja2):
     atparse_to_jinja2.convert(input_file=atparsefile)
     streams = capsys.readouterr()
     assert not streams.err
-    assert streams.out == jinja2txt
+    assert streams.out.strip() == txt_jinja2
+
+
+def test_convert_preserve_whitespace(tmp_path):
```

Comment:
Here's a new unit test to lock down expected behavior w.r.t. the reported bug.

Reply:
I updated this to include intentional leading and trailing newlines, which should also be preserved in the translation.